src/pwr.ps1

```diff
@@ -1,1426 +1,260 @@
-				foreach ($Name in $Pkgs.Keys) {
-					foreach ($Ver in $Pkgs.$Name) {
-						$PkgRoot = "$PwrPkgPath\$Name-$Ver"
-						try {
-							$Item = Get-ChildItem -Path "$PkgRoot\.pwr"
-						} catch {
-							Write-PwrWarning "$PkgRoot is not a pwr package; it should be removed manually"
-							continue
-						}
-						$Item.LastAccessTime = $Item.LastAccessTime
-						$Old = $Item.LastAccessTime -lt ((Get-Date) - (New-TimeSpan -Days $DaysOld))
-						if ($Old -and $PSCmdlet.ShouldProcess("${Name}:$Ver", 'remove pwr package')) {
-							Write-PwrOutput "removing ${Name}:$Ver ... " -NoNewline
-							Remove-Directory $PkgRoot
-							Write-PwrHost 'done'
-						}
-					}
-				}
-			} else {
-				Assert-NonEmptyPwrPackages
-				foreach ($P in $Packages) {
-					$Pkg = Assert-PwrPackage $P
-					if ($Pkg.Local) {
-						Write-PwrFatal "tried to remove local package $($Pkg.Ref)"
-					} elseif ("$env:PwrLoadedPackages".Contains($Pkg.Ref)) {
-						Write-PwrFatal "tried to remove loaded package $($Pkg.Ref)"
-					} elseif ("$env:PwrShellPackages".Contains($Pkg.Ref)) {
-						Write-PwrFatal "tried to remove shell session package $($Pkg.Ref)"
-					} elseif (Test-PwrPackage $Pkg) {
-						if ($PSCmdlet.ShouldProcess($Pkg.Ref, 'remove pwr package')) {
-							Write-PwrOutput "removing $($Pkg.Ref) ... " -NoNewline
-							$Path = Resolve-PwrPackage $Pkg
-							Remove-Directory $Path
-							Write-PwrHost 'done'
-						}
-					} else {
-						Write-PwrFatal "$($Pkg.Ref) not found"
-					}
-				}
-			}
-		} finally {
-			Unlock-PwrLock
-		}
-	}
-	'which' {
-		Assert-NonEmptyPwrPackages
-		foreach ($local:P in $Packages) {
-			$local:Pkg = Assert-PwrPackage $local:P
-			Write-PwrHost "$($local:Pkg.Ref) $($local:Pkg.Digest)"
+Set-Alias -Name 'airpower' -Value 'Invoke-Airpower' -Scope Global
+Set-Alias -Name 'air' -Value 'Invoke-Airpower' -Scope Global
+Set-Alias -Name 'pwr' -Value 'Invoke-Airpower' -Scope Global
+
+& {
+	if ('Airpower.psm1' -eq (Split-Path $MyInvocation.ScriptName -Leaf)) {
+		# Invoked as a module
+		$params = @{
+			URL = "https://www.powershellgallery.com/packages/airpower"
+			Method = 'HEAD'
 		}
-	}
-	'where' {
-		Assert-NonEmptyPwrPackages
-		foreach ($local:P in $Packages) {
-			$local:Pkg = Assert-PwrPackage $local:P
-			if (-not (Test-PwrPackage $local:Pkg)) {
-				Write-PwrFatal "package $($local:Pkg.Ref) is not installed"
+		$resp = HttpRequest @params | HttpSend -NoRedirect
+		if ($resp.Headers.Location) {
+			$remote = [Version]::new($resp.Headers.Location.OriginalString.Substring('/packages/airpower/'.Length))
+			$local = [Version]::new((Import-PowerShellDataFile -Path "$PSScriptRoot\Airpower.psd1").ModuleVersion)
+			if ($remote -gt $local) {
+				WriteHost "$([char]27)[92mA new version of Airpower is available! [v$remote]$([char]27)[0m"
 			}
-			Write-PwrHost (Resolve-PwrPackage $local:Pkg)
 		}
-	}
-	Default {
-		Write-PwrFatal "no such command '$Command'"
+		PrunePackages -Auto
 	}
 }
```

Comment:
As written, it runs on module load, so once per shell in which you make an `airpower` call.

If this feels too slow as you use it, we can reinstitute the "daily" cache to prevent this from always running.

Reply:
I wouldn't think this check would be significant, since it only needs to load the local database. But, I'll let you know if I notice anything. I'll also note that I reviewed the implementation, and I'm pretty sure it meets all of my wants for auto-pruning.